ci: bump versions of K3s/RKE2

ldevulder opened the pull request:
According to [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-9-1/) we should use K3s/RKE2 version 1.27 to 1.30.

Verification run:
- [CLI-K3s-OBS_Dev](https://github.com/rancher/elemental/actions/runs/11028861895)
- [CLI-RKE2-OBS_Dev](https://github.com/rancher/elemental/actions/runs/11031322789)
- [UI-K3s-OBS_Dev](https://github.com/rancher/elemental/actions/runs/11028869269)
- [UI-RKE2-OBS_Dev](https://github.com/rancher/elemental/actions/runs/11033333078)

ldevulder committed:
ci: bump versions of K3s/RKE2

Signed-off-by: Loic Devulder <ldevulder@suse.com>

juadk commented on .github/workflows/ui-rm-head-2.7-matrix.yaml:
```diff
@@ -14,11 +14,11 @@ on:
       k8s_upstream_version:
         description: Rancher cluster upstream version to use
-        default: '"v1.27.13+k3s1"'
+        default: '"v1.30.4+k3s1"'
```
In rancher 2.7-head, there is nothing higher than 1.27 for the kubernetes downstream cluster:
![image](https://github.com/user-attachments/assets/0f39e50a-2a95-494a-926b-e87dc8b653e9)

ldevulder replied:
Yes I saw it, fixed!

juadk commented on .github/workflows/ui-rm-head-2.7-matrix.yaml:
```diff
@@ -42,8 +42,8 @@ jobs:
-        k8s_downstream_version: ${{ fromJSON(format('[{0}]', inputs.k8s_downstream_version || '"v1.27.13+k3s1","v1.27.13+rke2r1"')) }}
-        k8s_upstream_version: ${{ fromJSON(format('[{0}]', inputs.k8s_upstream_version || '"v1.27.13+k3s1"')) }}
+        k8s_downstream_version: ${{ fromJSON(format('[{0}]', inputs.k8s_downstream_version || '"v1.30.4+k3s1","v1.30.4+rke2r1"')) }}
+        k8s_upstream_version: ${{ fromJSON(format('[{0}]', inputs.k8s_upstream_version || '"v1.30.4+k3s1"')) }}
```
Same here for the kubernetes downstream cluster version.

ldevulder replied:
Yes I saw it, fixed!

ldevulder committed:
ci/cli: move Rancher 2.8-head tests to weekly

As Rancher Prime and Community Edition are now both on 2.9.x version we
can move 2.8-head tests from daily to weekly.

Signed-off-by: Loic Devulder <ldevulder@suse.com>
